### PR TITLE
🎲 Fixes some tab tray stuffs

### LINKED ISSUE #10848: FNX-4625 ⁃ Update tab tray design
Please update the design to match the Abstract file (you can use Inspect mode to get values) https://share.goabstract.com/1fade20f-700b-4125-baf5-dca4ffd942ee

Please reach out for any questions, feedback or if a value seems off.

* [ ] Add full width separator between tab tray header and first open tab (1dp, light_grey_30)
* [ ] Keep rounded corners at top when expanded full
* [ ] Add white/dark BG for Android bottom bar (currently it has opacity)
* [ ] Update multi-task, private, 3-dot icon and hairline in light mode to ink_20 (#312A65) and in dark mode to violet_50 (#9059FF)
* [ ] reduce opacity of inactive non-private/private icon in light mode to ink_20_48a (#7A312A65)in dark mode to violet_50_48a (#7A9059FF)
* [ ] update new tab button in PBM: add label and update color in light mode to gradient purple_70 (#722291) to violet_80 (#45278D) and in dark mode to gradient purple_50 (#B833E1), violet_50 (#9059FF)

### PR DESCRIPTION
**Pull request by boek**
Fixes #10848, #11090

### CODE REVIEW

File: app/src/main/res/layout/component_tabstray_fab.xml [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
-<com.google.android.material.floatingactionbutton.FloatingActionButton
-    xmlns:android="http://schemas.android.com/apk/res/android"
+<com.google.android.material.floatingactionbutton.ExtendedFloatingActionButton xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
     android:id="@+id/new_tab_button"
+    style="@style/Widget.MaterialComponents.ExtendedFloatingActionButton"
     android:layout_width="wrap_content"
     android:layout_height="wrap_content"
-    android:layout_gravity="bottom|end"
+    android:layout_gravity="bottom|right"
```
Comment (darkwing) [inline app/src/main/res/layout/component_tabstray_fab.xml:11]: Could this be an RTL problem?

File: app/src/main/res/layout/component_tabstray_fab.xml [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
     android:layout_margin="16dp"
-    android:contentDescription="@string/add_tab"
-    app:backgroundTint="@color/accent_normal_theme"
-    app:srcCompat="@drawable/ic_new"
-    app:tint="@color/photonWhite" />
+    android:backgroundTint="@color/accent_normal_theme"
+    android:contentDescription="this is also a test"
+    android:elevation="99dp"
+    android:text="PRIVATE"
```
Comment (darkwing) [inline app/src/main/res/layout/component_tabstray_fab.xml:16]: We should add these strings for L10N.

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt [hunk 3]
```diff
@@ -204,12 +212,21 @@ class TabTrayView(
     override fun onTabReselected(tab: TabLayout.Tab?) { /*noop*/ }
     override fun onTabUnselected(tab: TabLayout.Tab?) { /*noop*/ }
 
+    fun toggleFabText(private: Boolean) {
```
Comment (darkwing) [inline app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt:215]: 🥇 🥇 🥇 🥇 🥇

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt [hunk 4]
```diff
@@ -95,6 +98,10 @@ class TabTrayView(
 
         view.tab_layout.addOnTabSelectedListener(this)
 
+        // We need to find a fix for this, this will require figuring out why the extended FAB won't
+        // respect any new elevation values
+        view.tab_wrapper.elevation = TEMP_ELEVATION
```
Comment (darkwing) [inline app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt:103]: What's the issue here?  Is the FAB displaying behind the tray or something?

**Comment by darkwing** [comment]
This is really good.  I think there are a few small side effects; for example, if I close my last private tab, the tab tray switches to my default tabs and the default home loads; I think I'd expect the private home to display.  In any event, this is a *massive* step forward.

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt (hunk 4 above)
Comment (boek) [reply app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt:103]: Yes

**Comment by boek** [comment]
ok I will try to fix the elevation issue and nits here before we land 👍

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -436,6 +436,10 @@
     <string name="tabs_header_private_tabs_title">Private tabs</string>
     <!-- Content description (not visible, for screen readers etc.): Add tab button. Adds a news tab when pressed -->
     <string name="add_tab">Add tab</string>
+    <!-- Content description (not visible, for screen readers etc.): Add tab button. Adds a news tab when pressed -->
+    <string name="add_private_tab">Add private tab</string>
+    <!-- Content for FAB when looking at private tabs in the tab drawer -->
```
Comment (flodolo) [inline app/src/main/res/values/strings.xml:441]: Hi, can someone update this comment to explain what FAB stands for, if it's relevant for localizers?

We're starting to have a bit too many strings hold by this question
https://github.com/mozilla-l10n/android-l10n/pull/225/files

**Pull request merged**